syntax = "proto3";

package PbTelemetry;

message MessagesPerSecond
{
    string deviceName = 1;
    uint64 bitsPerSecond = 2;
    double busLoad = 3;
    uint64 count = 4;
}

message Camera
{
    string status = 1;
    string error = 2;
}

message Status
{
    uint64 timestamp = 1;
    uint64 zeroTimestamp = 2;
    int64 data = 3;
    uint64 cpuTotalLoad = 4;
    uint64 cpuProcessLoad = 5;
    uint64 memProcessLoad = 6;
    uint64 canlibBuildTime = 7;
    uint64 telemetryBuildTime = 8;
    Camera camera = 9;
    repeated MessagesPerSecond messagesPerSecond = 10;
}